x-pack/plugins/timelines/server/search_strategy/timeline/factory/helpers/build_object_recursive.ts

```diff
@@ -6,13 +6,13 @@
  */
 
 import { set } from '@elastic/safer-lodash-set';
-import { get, has } from 'lodash/fp';
+import { get } from 'lodash/fp';
 import { Ecs } from '../../../../../common/ecs';
-import { EventHit, Fields } from '../../../../../common/search_strategy';
+import { Fields } from '../../../../../common/search_strategy';
 import { toStringArray } from '../../../../../common/utils/to_array';
 import { getNestedParentPath } from './get_nested_parent_path';
 
-const buildObjectRecursive = (fieldPath: string, fields: Fields): Partial<Ecs> => {
+export const buildObjectRecursive = (fieldPath: string, fields: Fields): Partial<Ecs> => {
```

Comment:
Question, why can't this function just be:

```
const buildObjectRecursive = (fieldPath: string, fields: Fields): Partial<Ecs> => {
  return set({}, fieldPath, toStringArray(get(fieldPath, fields)));
};
```

Reply:
This function needs to be recursive, because we need to rebuild the objects from the nested structure of the field path, I've tried the code you've suggested, but it doesn't build the object properly. I didn't find any simpler way to do this.